internal/loader/settings_client.go

```diff
@@ -72,15 +133,19 @@ func (s *SelectorSettingsClient) GetSettings(ctx context.Context, client *azappc
 				Fields:      azappconfig.AllSettingFields(),
 			}
 			pager := client.NewListSettingsPager(selector, nil)
+			latestEtags := make([]*azcore.ETag, 0)
 
 			for pager.More() {
 				page, err := pager.NextPage(ctx)
 				if err != nil {
 					return nil, err
 				} else if len(page.Settings) > 0 {
```

Comment:
Would this check be problematic? What if it returns an empty setting list?

Reply:
Yes, it should check if `page.Settings == nil`, when empty setting returned, `pageEtag` is not empty.